peerprep-backend/src/main/java/com/peerprep/peerprepbackend/dto/request/CreateUserRequest.java

```diff
@@ -0,0 +1,13 @@
+package com.peerprep.peerprepbackend.dto.request;
+
+import lombok.NoArgsConstructor;
+import lombok.Value;
+
+@Value
+@NoArgsConstructor(force = true)
+public class CreateUserRequest {
+    String username;
+    String email;
+    String country;
+    String password;
```

Comment:
If im not wrong request object fields are optional by default (can test abit, it worked for me), but in any case use spring's validation to enforce `@NotNull` at the controller level. Forgot to add it for question too..
https://www.baeldung.com/spring-boot-bean-validation

Reply:
yep, done! thanks! uhh I tried to add for question, but the complexity and categories low key kinda pain bc they're enums so here's a possible resource :'D

https://www.baeldung.com/javax-validations-enums

Reply:
same for this, i think `@NotBlank` might be wrong cos it still allows null values, @NotEmpty should be good.
I'll try to implement for question, thanks!